### PR TITLE
fix: Keep user service in server, instead of creating

### PR DESCRIPTION
**Pull request by gaocegege**
Ref #110 

This PR is to:

- Refactor user service

Signed-off-by: Ce Gao <cegao@tensorchord.ai>

### CODE REVIEW

**Comment by gaocegege** [comment]
PTAL @kemingy @VoVAllen

File: pkg/service/user/user.go [hunk 1]
```diff
@@ -15,20 +15,29 @@ import (
...
+type generalService struct {
 	querier   query.Querier
 	jwtIssuer *JWTIssuer
 }
 
 func NewService(querier query.Querier,
-	secret string, expirationTimeDefault time.Duration) *Service {
-	return &Service{
+	secret string, expirationTimeDefault time.Duration) Service {
+	return &generalService{
```
Comment (kemingy) [inline pkg/service/user/user.go:34]: So it's a pointer but then convert into the value?

Comment (gaocegege) [reply pkg/service/user/user.go:34]: It's the pointer to the struct `generalService`, which implements the interface `Service`

Comment (kemingy) [reply pkg/service/user/user.go:34]: I see. It has to be a pointer so the compiler will know whether it implemented the interface or not. Golang syntax is weird.

**Review by kemingy — APPROVED** [review]

File: pkg/service/user/user.go [hunk 2]
```diff
@@ -15,20 +15,29 @@ import (
 	"github.com/tensorchord/envd-server/pkg/query"
 )
 
-type Service struct {
+type Service interface {
+	Register(loginName, pwd string,
+		PublicKey []byte) (string, error)
```
Comment (xieydd) [inline pkg/service/user/user.go:20]: PublicKey -> publicKey

**Pull request merged**